apps/examples/testcase/le_tc/kernel/itc_semaphore.c

```diff
@@ -0,0 +1,67 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file itc_semaphore.c
+/// @brief Test Case Example for semaphore API
+#include <tinyara/config.h>
+#include <semaphore.h>
+#include <pthread.h>
+#include "tc_internal.h"
+#include <stdio.h>
+#include <sys/types.h>
+
+#define LOOP_SIZE CONFIG_ITC_KERN_SEM_LOOP_SIZE
+#define PSHARED 0
+#define SEM_COUNT 1
+
+/**
+* @fn                   :itc_semaphore_sem_init_destroy_p_multitime
+* @brief                :sem_destroy destroys semaphore
+* @scenario             :sem init destroy multitime
+* API's covered         :sem_init,sem_destroy
+* Preconditions         :none
+* Postconditions        :none
+*/
+static void itc_semaphore_sem_init_destroy_p_multitime(void)
+{
+	sem_t sem;
+	int ret_chk;
+	int exec_index;
+
+	for (exec_index = 0; exec_index < LOOP_SIZE; exec_index++) {
```

Comment:
This loop only checks the pair of sem init-destroy. You'd better check another case for example, trying to destroy the destroyed sem.

Reply:
As per discussion related to other comment in same file with Mr sunghan-chang and Linux Manual  
Destroying a semaphore that other processes or threads are currently
blocked on (in sem_wait(3)) produces undefined behavior. thats why destrying alraedy destroy semaphore doesnt produce any result.